src/cpu.c

```diff
@@ -434,6 +459,28 @@ static void aggregate (gauge_t *sum_by_state) /* {{{ */
 
 		RATE_ADD (sum_by_state[COLLECTD_CPU_STATE_ACTIVE], this_cpu_states[COLLECTD_CPU_STATE_ACTIVE].rate);
 	}
+
+#if defined(HAVE_PERFSTAT) /* {{{ */
+	cdtime_t now = cdtime ();
+	perfstat_cpu_total_t cputotal;
+	memset(&cputotal, 0, sizeof(perfstat_cpu_total_t));
```

Comment:
If possible, please initialize `cputotal` using C's struct initializer:

```c
perfstat_cpu_total_t cputotal = { 0 };
```

If the first field of the struct is another struct, GCC may misbehave. In that case, please use something like this:

```c
perfstat_cpu_total_t cputotal = { .pidle = 0 };
```